### ISSUE TITLE
Incorrect magit diff inheritage

### ISSUE THREAD

**Issue by renatofdds** [bbatsov/zenburn-emacs#305]
There is no difference between diff refines on current hunk on magit because magit highlight faces are set to inherit from diff-refines:

```
`(magit-diff-added ((t (:inherit diff-added))))
`(magit-diff-added-highlight ((t (:inherit diff-refine-added))))
`(magit-diff-removed ((t (:inherit diff-removed))))
`(magit-diff-removed-highlight ((t (:inherit diff-refine-removed))))
```

Which is incorrect:
```magit-diff-added-highlight``` should be used to highlight the *entire selected hunk* (and not refinements) as it gets applied to the entire currently selected hunk checked by  ```(magit-section-selected-p section selection)```. Therefore ```diff-refine-added``` should be distinguished from ```magit-diff-added-highlight```. 

I suggest: 
```
`(diff-refine-added   ((t (:background ,zenburn-green-3 :foreground ,zenburn-green+4))))
`(magit-diff-added-highlight   ((t (:background ,zenburn-green-4 :foreground ,zenburn-green+2))))

`(diff-refine-removed ((t (:background ,zenburn-red-4 :foreground ,zenburn-red+3))))
`(magit-diff-removed-highlight ((t (:background ,zenburn-red-5 :foreground ,zenburn-red+1))))
```

**Comment by tarsius** [comment]
Fixed in #317.

**Comment by impaktor** [comment]
That's now been merged, so this can be closed.
